### PR TITLE
Updated code according Swift v.3

### PR DESCRIPTION
**Pull request by Rogaven**
- [✔] PR touches only one file (or a set of logically related files with similar changes made)
- [✔] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [✔] YAML Frontmatter formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
- [✔] Seriously, look at it now. Watch for quotes and double-check field names.

### CODE REVIEW

File: swift.html.markdown [hunk 1]
```diff
@@ -104,10 +105,13 @@ if let someOptionalStringConstant = someOptionalString {
...
-// AnyObject == id
-// Unlike Objective-C `id`, AnyObject works with any value (Class, Int, struct, etc.)
-var anyObjectVar: AnyObject = 7
-anyObjectVar = "Changed value to a string, not good practice, but possible."
+// Any == id
+// Unlike Objective-C `id`, Any works with any value (Class, Int, struct, etc.)
+var anyVar: Any = 7
+anyVar = "Changed value to a string, not good practice, but possible."
+
+// Also there is similar type – `AnyObject`, which works only for objects
```
Comment (vendethiel) [inline swift.html.markdown]: So in the end, `id == AnyObject`, and the original version was right?

File: swift.html.markdown [hunk 2]
```diff
@@ -151,11 +155,14 @@ var explicitEmptyMutableDictionary: [String: Float] = [:] // same as above
 // MARK: Control Flow
 //
 
-// Condition statements support "where" clauses, which can be used
-// to help provide conditions on optional values.
-// Both the assignment and the "where" clause must pass.
+/* ~~Condition statements support "where" clauses, which can be used
```
Comment (vendethiel) [inline swift.html.markdown]: I think you can remove the quotes now that it's not a keyword anymore

File: swift.html.markdown [hunk 3]
```diff
@@ -289,14 +296,14 @@ var increment = makeIncrementer()
 increment(7)
 
 // pass by ref
-func swapTwoInts(inout a: Int, inout b: Int) {
+func swapTwoInts( a: inout Int, b: inout Int) {
```
Comment (vendethiel) [inline swift.html.markdown]: spurious space before "a"

File: swift.html.markdown [hunk 4]
```diff
@@ -572,8 +579,8 @@ protocol ShapeGenerator {
 // Protocols declared with @objc allow optional functions,
 // which allow you to check for conformance
```
Comment (vendethiel) [inline swift.html.markdown]: please add a note about the fact functions need it as well

File: swift.html.markdown [hunk 5]
```diff
@@ -657,8 +664,8 @@ print(mySquare.sideLength) // 4
 print(mySquare.sideLength) // 12
 
 // Operators can also be generics
-infix operator <-> {}
-func <-><T: Equatable> (inout a: T, inout b: T) {
+infix operator <->
+func <-><T: Equatable> ( a: inout T, b: inout T) {
```
Comment (vendethiel) [inline swift.html.markdown]: spurious space before "a"

Comment (Rogaven) [inline swift.html.markdown]: Missed that

File: swift.html.markdown (hunk 2 above)
Comment (Rogaven) [inline swift.html.markdown]: Yep, will remove

File: swift.html.markdown (hunk 4 above)
Comment (Rogaven) [inline swift.html.markdown]: Ok

File: swift.html.markdown (hunk 1 above)
Comment (Rogaven) [inline swift.html.markdown]: I'll rewrite this. I mean 
`AnyObject` == `id` from Objective-C
`Any` – also works with any scalar values (Class, Int, struct, etc.)
And in current version of Swift we can't do such things `var anyObjectVar: AnyObject = 7`

**Comment by vendethiel** [comment]
Thanks!

**Pull request merged**